packages/next/src/build/webpack/loaders/next-app-loader.ts

```diff
@@ -634,20 +636,15 @@ const nextAppLoader: AppLoader = async function nextAppLoader() {
   // This can be more efficient than checking them with `fs.stat` one by one
   // because all the thousands of files are likely in a few possible directories.
   // Note that it should only be cached for this compilation, not globally.
-  const filesInDir = new Map<string, Set<string>>()
+  const filesInDir = new Map<string, Promise<Set<string>>>()
   const fileExistsInDirectory = async (dirname: string, fileName: string) => {
-    const existingFiles = filesInDir.get(dirname)
-    if (existingFiles) {
-      return existingFiles.has(fileName)
-    }
-    try {
-      const files = await getFilesInDir(dirname)
-      const fileNames = new Set<string>(files)
-      filesInDir.set(dirname, fileNames)
-      return fileNames.has(fileName)
-    } catch (err) {
-      return false
+    if (!filesInDir.has(dirname)) {
+      filesInDir.set(
+        dirname,
+        getFilesInDir(dirname).catch(() => new Set())
```

Comment:
We aren't using `graceful-fs` or webpack's `intermediateFileSystem` I think it's possible that we throw here and bugger a build, but previously we'd also bugger a build on error. We would just would do it for fewer entries as the error would prevent caching.

Reply:
If an error occurs here I expect it to bubble and fail the build, I don't think we should silently hide this as an empty result

Reply:
I was just trying to match the existing behavior.

I don't what the intention was w/ the try/catch...  I could see either:
 - we're checking against a possible list, of which we may expect a directory not to be there
 - we've tried to guard against too many fs calls as we're not using graceful-fs/webpack fs here.

Reply:
Since we weren't precise with the error before I guess this is fine although we be good for us to explicitly check it was an `ENOTFOUND` or something